go/worker/compute/executor/committee/node.go

```diff
@@ -729,25 +730,40 @@ func (n *Node) proposeTimeoutLocked(roundCtx context.Context) error {
 	return nil
 }
 
-func (n *Node) getRtStateAndRoundResults(ctx context.Context, height int64) (*roothash.RuntimeState, *roothash.RoundResults, error) {
+func (n *Node) getRuntimeRoundData(ctx context.Context, height int64, currentRound uint64) (*roothash.RuntimeState, *roothash.RoundResults, []*staking.Event, error) {
 	state, err := n.commonNode.Consensus.RootHash().GetRuntimeState(ctx, n.commonNode.Runtime.ID(), height)
 	if err != nil {
 		n.logger.Error("failed to query runtime state",
 			"err", err,
 			"height", height,
 		)
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	roundResults, err := n.commonNode.Runtime.History().GetRoundResults(ctx, state.LastNormalRound)
 	if err != nil {
 		n.logger.Error("failed to query round last normal round results",
 			"err", err,
 			"height", height,
 		)
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	return state, roundResults, nil
+	// Load all events since last successful round.
+	var stakingEvents []*staking.Event
+	for r := state.LastNormalRound; r < currentRound; r++ {
```

Comment:
Hm so round `r` will only include events that happened between last normal round and last round? I guess this makes sense as we need a consistent snapshot and until there's a round we don't have one, but we need to document it as this makes events slightly delayed.

Reply:
> I guess this makes sense as we need a consistent snapshot and until there's a round we don't have one

Yeah this, but also this way the consensus events match the consensus state the runtime gets. (`CurrentConsensusBlock` that is passed to the runtime is from the height at the previous round). So this makes the events and consensus state the runtime sees consistent. I think having these not in sync would be more confusing.

Reply:
> So this makes the events and consensus state the runtime sees consistent. 

Yes that's right (we should add a test to make sure). I'm not saying that this should be changed, it should just be documented.